ironic/tests/unit/drivers/modules/drac/test_inspect.py

```diff
@@ -378,3 +387,81 @@ def test__get_pxe_dev_nics_with_empty_list(self, mock_get_drac_client):
+            'PxeDev1EnDis': 'Enabled', 'PxeDev2EnDis': 'Disabled',
+            'PxeDev3EnDis': 'Disabled', 'PxeDev4EnDis': 'Disabled',
+            'PxeDev1Interface': 'NIC.Integrated.1-1-1',
+            'PxeDev2Interface': None, 'PxeDev3Interface': None,
+            'PxeDev4Interface': None}
+
+        return system_mock
+
+    def test_get_properties(self):
+        expected = redfish_utils.COMMON_PROPERTIES
+        driver = drac_inspect.DracRedfishInspect()
+        self.assertEqual(expected, driver.get_properties())
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    def test__get_pxe_dev_nics_with_UEFI_boot_mode(self, mock_get_system):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = 'uefi'
+        expected_pxe_nic = ['24:6E:96:70:49:00']
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    def test__get_pxe_dev_nics_with_BIOS_boot_mode(self, mock_get_system):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = 'bios'
+        expected_pxe_nic = ['24:6E:96:70:49:04']
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    def test__get_pxe_dev_nics_without_boot_mode(self, mock_get_system):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = None
+        expected_pxe_nic = []
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    @mock.patch.object(DracRedfishInspect, '_get_pxe_dev_nics', autospec=True)
+    @mock.patch.object(DracRedfishInspect, 'inspect_hardware', autospec=True)
+    def test_inspect_hardware_without_pxe_dev_nics(
+            self, mock_get_system, mock__get_pxe_dev_nics,
+            mock_inspect_hardware):
+        mock__get_pxe_dev_nics.return_value = None
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(
+                task)
+            self.assertIsNone(pxe_dev_nics)
+            self.assertEqual(1, mock__get_pxe_dev_nics.call_count)
```

Comment:
Can you add a test that validates that pxe_enabled is set correctly on the ports on the node?

Reply:
Done

Reply:
Done